### PR TITLE
move transactional to implementation

### PR DESCRIPTION
**Pull request by AntonStepanof**
fixes https://github.com/OHDSI/Atlas/issues/1879

### CODE REVIEW

File: src/main/java/org/ohdsi/webapi/service/IRAnalysisService.java [hunk 1]
```diff
@@ -572,6 +580,7 @@ public IRAnalysisDTO copy(final int id) {
 
 
   @Override
+  @Transactional
   public Response export(final int id) {
```
Comment (wivern) [inline src/main/java/org/ohdsi/webapi/service/IRAnalysisService.java]: I'd like to suggest add source access checking and filtering during export.

Comment (wivern) [reply src/main/java/org/ohdsi/webapi/service/IRAnalysisService.java]: In addition to above comment, there is an issue when some of sources became unavailable. During such case export analysis to CSV also becomes unavailable. Probably, it's better to export results for working sources and notify a user that some sources doesn't responsible.

Comment (AntonStepanof) [reply src/main/java/org/ohdsi/webapi/service/IRAnalysisService.java]: I added check for availability sources

Comment (AntonStepanof) [reply src/main/java/org/ohdsi/webapi/service/IRAnalysisService.java]: also added check for permissions

File: src/main/java/org/ohdsi/webapi/service/IRAnalysisService.java [hunk 2]
```diff
@@ -785,4 +799,19 @@ private void fillCohortIds(List<Integer> ids, List<CohortDTO> cohortDTOS) {
...
+
+  private boolean isSourceAvailable(Source source) {
+    boolean sourceAvailable = true;
+    try {
+      sourceService.checkConnection(source.getSourceKey());
+    } catch (Exception e) {
+      log.error("cannot get connection to source with key {}", source.getSourceKey(), e);
+      sourceAvailable = false;
+    }
+    if (!sourceAccessor.hasAccess(source)) {
```
Comment (wivern) [inline src/main/java/org/ohdsi/webapi/service/IRAnalysisService.java]: Not needed to detect connection availability when source access is prohibited, don't it?

File: src/main/java/org/ohdsi/webapi/service/IRAnalysisService.java [hunk 3]
```diff
@@ -596,6 +605,9 @@ public Response export(final int id) {
       for (ExecutionInfo execution : executions)
       {
         Source source = execution.getSource();
+        if (!isSourceAvailable(source)) {
```
Comment (wivern) [inline src/main/java/org/ohdsi/webapi/service/IRAnalysisService.java]: But why not simply filter out executions with unaccessible source?
Smth like `executions.stream().filter(e -> this.isSourceAvailable(e.getSource()).collect(...)` ?

File: src/main/java/org/ohdsi/webapi/service/IRAnalysisService.java (hunk 2 above)
Comment (AntonStepanof) [reply src/main/java/org/ohdsi/webapi/service/IRAnalysisService.java]: yes, fixed

Comment (AntonStepanof) [reply src/main/java/org/ohdsi/webapi/service/IRAnalysisService.java]: yes, updated

File: src/main/java/org/ohdsi/webapi/service/IRAnalysisService.java (hunk 3 above)
Comment (AntonStepanof) [reply src/main/java/org/ohdsi/webapi/service/IRAnalysisService.java]: updated

**Comment by wivern** [comment]
dceab168-aded-4d78-b98c-4bf7a79ed0b8
idea

**Pull request merged**